sticker/src/tensorflow/dataset.rs

```diff
@@ -45,6 +46,27 @@ impl<R> ConllxDataSet<R> {
     pub fn new(read: R) -> Self {
         ConllxDataSet(read)
     }
+
+    /// Returns an `Iterator` over `Result<Sentence, Error>`.
+    ///
+    /// Depending on the parameters the returned iterator filters
+    /// sentences by their lengths or returns the sentences in
+    /// sequence without filtering them.
+    ///
+    /// If `max_len` == `usize::MAX`, no filtering is performed.
+    fn get_sentence_iter<'a>(
+        reader: R,
+        max_len: usize,
+    ) -> Box<dyn Iterator<Item = Result<Sentence, Error>> + 'a>
+    where
+        R: ReadSentence + 'a,
```

Comment:
There are two options here:

Option a) is the current state, we don't have a local parameter on `get_sentence_iter` but use `R` from `ConllxDataSet`.

Option b) introduces a local parameter `S` on `get_sentence_iter` and needs to specify the type of the `R` on `ConllxDataSet` when calling `get_sentence_iter`:

```rust
    fn get_sentence_iter<'a,S>(
        reader: S,
        max_len: usize,
    ) -> Box<dyn Iterator<Item = Result<Sentence, Error>> + 'a>
    where
        S: ReadSentence + 'a,
    {...}

  
    fn batches(...) -> Fallible<Self::Iter> {

        Ok(ConllxIter {
            batch_size,
            encoder,
            sentences: ConllxDataSet::<R>::get_sentence_iter(reader, max_len),
            vectorizer,
        })
    }
}
```

Reply:
I think the current approach is good enough. We could always relax the type parameters when necessary.